lib/blocs/request_manager/online_request_manager.dart

```diff
@@ -110,22 +117,24 @@ class OnlineRequestManager implements RequestManager {
   }
 
   Future<Plan> _fetchPlan(
-    BuildContext context,
-    TrufiLocation from,
-    TrufiLocation to,
-    String mode,
-  ) async {
+      BuildContext context, TrufiLocation from, TrufiLocation to, String mode,
+      [Map<String, String> additionalParams]) async {
     final preferences = PreferencesBloc.of(context);
-    Uri request = Uri.parse(
-      TrufiConfiguration().url.otpEndpoint + planPath,
-    ).replace(queryParameters: {
+    final params = {
       "fromPlace": from.toString(),
       "toPlace": to.toString(),
-      "date": _todayMonthDayYear(),
+      "date": _todayOtpDate(),
+      "time": _todayOtpTime(),
       "numItineraries": "5",
       "mode": mode,
       "correlation": preferences.correlationId,
-    });
+    };
+    if (additionalParams != null) {
+      params.addAll(additionalParams);
+    }
```

Comment:
```dart
    final params = {
      "fromPlace": from.toString(),
      "toPlace": to.toString(),
      "date": _todayOtpDate(),
      "time": _todayOtpTime(),
      "numItineraries": "5",
      "mode": mode,
      "correlation": preferences.correlationId,
      if (additionalParams != null) ...additionalParams
    };
```

Reply:
Haha, true. IIRC that feature was introduced to Dart after this code was written :)